sbi/utils/tensorboard_output.py

```diff
@@ -0,0 +1,213 @@
+"""Utils for processing tensorboard event data."""
+import inspect
+import logging
+from copy import deepcopy
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+from tensorboard.backend.event_processing.event_accumulator import (
+    DEFAULT_SIZE_GUIDANCE,
+    EventAccumulator,
+)
+
+from sbi.utils.plot import _get_default_opts
+from sbi.utils.io import get_log_root
+import sbi.inference.base
+
+# creating an alias for annotating, because sbi.inference.base.NeuralInference creates
+# a circular import error
+_NeuralInference = Any
+
+
+def plot_summary(
+    inference: Union[_NeuralInference, Path],
+    tags: List[str] = ["validation_log_probs_across_rounds"],
+    disable_tensorboard_prompt: bool = False,
+    tensorboard_scalar_limit: int = 10_000,
+    figsize: List[int] = [20, 6],
+    fontsize: float = 12,
+    fig: Optional[Figure] = None,
+    axes: Optional[Axes] = None,
+    xlabel: str = "epochs",
+    ylabel: List[str] = [],
+    plot_kwargs: Dict[str, Any] = {},
+) -> Tuple[Figure, Axes]:
+    """Plots tensorboard scalar events logged by the summary writer held by an
+     inference object.
```

Comment:
Nitpick: first line of docstring should be a single line [PEP 257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings)